### PR TITLE
`General`: Fix navigation highlighting for 'Assessment' link

### PR DESCRIPTION
**Pull request by DominikRemo**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] Language: I followed the [guidelines for inclusive, diversity-sensitive, and appreciative language](https://docs.artemis.cit.tum.de/dev/guidelines/language-guidelines/).
- [x] I chose a title conforming to the [naming conventions for pull requests](https://docs.artemis.cit.tum.de/dev/development-process/#naming-conventions-for-github-pull-requests).

#### Client
- [x] I followed the [coding and design guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client/).
- [x] I added multiple integration tests (Jest) related to the features (with a high test coverage), while following the [test guidelines](https://docs.artemis.cit.tum.de/dev/guidelines/client-tests/).
- [x] I documented the TypeScript code using JSDoc style.

### Motivation and Context
This PR fixes a bug in the course management tab bar. Before the fix, it was possible to have two links _(Exams & Assessment)_ highlighted at the same time when navigating to an exam's grading key.

### Description
The shouldHighlightAssessmentLink() function now excludes all links containing "exams".

### Steps for Testing
Prerequisites:
- 1 Instructor
- 1 Exam with grading key

1. Navigate to the exam grading key
2. Note that only the exam tab is highlighted
3. Make sure that the assessment links still work as expected

Additional:
- Navigate around and verify that the same bug does not appear in other places.

#### Code Review
- [x] Code Review 1
- [x] Code Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

#### Client

| Class/File | Line Coverage | Confirmation (assert/expect) |
|------------|--------------:|---------------------:|
| [course-management-tab-bar.component.ts](https://bamboo.ase.in.tum.de/browse/ARTEMIS-TESTS5300/latest/artifact/shared/Coverage-Report-Client-Tests/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts.html) | 96.77% | ✅ |

### Screenshots
Before the bugfix, the _exam_ and _assessment dashboard_ link were highlighted on the exam grading key page.
Bildschirmfoto 2023-08-11 um 07 22 24

### CODE REVIEW

**Review by chrisknedl — DISMISSED** [review]
Was able reproduce the issue locally on the develop branch, and can confirm that this PR fixes it (highlighting seems to be working as expected in other areas as well). Code also looks clean 👍

**Review by basak-akan — DISMISSED** [review]
The issue was reproduced on TS5 and the fix worked, only the exam tab is highlighted.
Could not detect the same issue appearing in other places.

**Review by laadvo — DISMISSED** [review]
Code changes look good to me :)

File: src/main/webapp/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts [hunk 1]
```diff
@@ -128,16 +128,31 @@ export class CourseManagementTabBarComponent implements OnInit, OnDestroy {
...
 
+    /**
+     * Checks if the current route contains 'grading-system' or 'plagiarism-cases' but not 'exams'.
+     * @return true if the current route is part of the assessment management
+     */
     shouldHighlightAssessmentLink(): boolean {
-        const assessmentLinkRegex = /grading-system|plagiarism-cases/;
-        return assessmentLinkRegex.test(this.router.url);
+        // Exclude exam related links from the assessment link highlighting
+        // example that should not highlight the assessment link: /course-management/{courseId}/exams/{examId}/grading-system/interval
```
Comment (nityanandaz) [inline src/main/webapp/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts:146]:
```suggestion
        // Exclude exam related links from the assessment link highlighting.
        // Example that should not highlight the assessment link: /course-management/{courseId}/exams/{examId}/grading-system/interval
```

File: src/main/webapp/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts [hunk 2]
```diff
@@ -128,16 +128,31 @@ export class CourseManagementTabBarComponent implements OnInit, OnDestroy {
...
+        const excludeRegex = /exams/;
+        const assessmentLinkRegex = /grading-system|plagiarism-cases|assessment-dashboard/;
+        return assessmentLinkRegex.test(this.router.url) && !excludeRegex.test(this.router.url);
     }
 
+    /**
+     * Checks if the current route is 'course-management/{courseId}/edit' or 'course-management/{courseId}'.
+     * @return true if the control buttons (e.g. delete & edit) should be shown
```
Comment (nityanandaz) [inline src/main/webapp/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts:154]:
```suggestion
     * @return true if the control buttons, e.g., delete & edit, should be shown
```

**Review by nityanandaz — COMMENTED** [review]
Minor grammar improvements

File: src/main/webapp/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts [hunk 3]
```diff
@@ -128,16 +128,31 @@ export class CourseManagementTabBarComponent implements OnInit, OnDestroy {
...
+     * @return true if the current route is part of the assessment management
+     */
     shouldHighlightAssessmentLink(): boolean {
-        const assessmentLinkRegex = /grading-system|plagiarism-cases/;
-        return assessmentLinkRegex.test(this.router.url);
+        // Exclude exam related links from the assessment link highlighting
+        // example that should not highlight the assessment link: /course-management/{courseId}/exams/{examId}/grading-system/interval
+        const excludeRegex = /exams/;
+        const assessmentLinkRegex = /grading-system|plagiarism-cases|assessment-dashboard/;
+        return assessmentLinkRegex.test(this.router.url) && !excludeRegex.test(this.router.url);
```
Comment (bassner) [inline src/main/webapp/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts:149]: What about using this pattern? You'd only have to run a single test then

```suggestion
        const assessmentLinkRegex = /^(?!.*exams).*(grading-system|plagiarism-cases|assessment-dashboard)/;
        return assessmentLinkRegex.test(this.router.url);
```

Comment (DominikRemo) [reply src/main/webapp/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts]: I tested it locally and it works. Thank you for your suggestion :)

File: src/main/webapp/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts (hunk 1 above)
Comment (DominikRemo) [reply src/main/webapp/app/course/manage/course-management-tab-bar/course-management-tab-bar.component.ts]: Thank you for your suggestions :)

**Review by laadvo — APPROVED** [review]
Reapprove code

**Review by RY997 — APPROVED** [review]
Worked well on ts1!

**Review by basak-akan — APPROVED** [review]
Tested on TS5 still works

**Review by chrisknedl — APPROVED** [review]
Re-approve

**Pull request merged**